static/app/views/detectors/components/detectorListTable/actions.tsx

```diff
@@ -0,0 +1,255 @@
+      `Are you sure you want to enable these %s monitors?`,
+      selected.size
+    );
+  }, [allInQuerySelected, queryCount, selected.size]);
+
+  const getDisableConfirmMessage = useCallback(() => {
+    if (allInQuerySelected) {
+      return tct(
+        'Are you sure you want to disable all [queryCount] monitors that match the search?',
+        {
+          queryCount,
+        }
+      );
+    }
+    return tn(
+      `Are you sure you want to disable this %s monitor?`,
+      `Are you sure you want to disable these %s monitors?`,
+      selected.size
+    );
+  }, [allInQuerySelected, queryCount, selected.size]);
+
+  const handleUpdate = useCallback(
+    ({enabled}: {enabled: boolean}) => {
+      openConfirmModal({
+        message: enabled ? getEnableConfirmMessage() : getDisableConfirmMessage(),
+        confirmText: enabled ? t('Enable') : t('Disable'),
+        priority: 'danger',
+        onConfirm: async () => {
+          if (allInQuerySelected) {
+            await updateDetectors({enabled, query, projects: selection.projects});
+          } else {
+            await updateDetectors({enabled, ids: Array.from(selected)});
+          }
+          togglePageSelected(false);
+        },
+      });
+    },
+    [
+      selected,
+      allInQuerySelected,
+      updateDetectors,
+      getEnableConfirmMessage,
+      getDisableConfirmMessage,
+      togglePageSelected,
+      selection.projects,
+      query,
+    ]
+  );
+
+  const getDeleteConfirmMessage = useCallback(() => {
+    if (allInQuerySelected) {
+      return tct(
+        'Are you sure you want to delete all [queryCount] monitors that match the search?',
+        {
+          queryCount,
+        }
+      );
+    }
+    return tn(
+      `Are you sure you want to delete this %s monitor?`,
+      `Are you sure you want to delete these %s monitors?`,
+      selected.size
+    );
+  }, [allInQuerySelected, queryCount, selected.size]);
+
+  const handleDelete = useCallback(() => {
+    openConfirmModal({
+      message: getDeleteConfirmMessage(),
+      confirmText: t('Delete'),
+      priority: 'danger',
+      onConfirm: async () => {
+        if (allInQuerySelected) {
+          await deleteDetectors({query, projects: selection.projects});
+        } else {
+          await deleteDetectors({ids: Array.from(selected)});
+        }
+        togglePageSelected(false);
+      },
+    });
+  }, [
+    selected,
+    allInQuerySelected,
+    deleteDetectors,
+    getDeleteConfirmMessage,
+    togglePageSelected,
+    selection.projects,
+    query,
+  ]);
+
+  return (
+    <Fragment>
+      <SimpleTable.Header>
+        <ActionsBarWrapper>
+          <Checkbox
+            checked={pageSelected || (anySelected ? 'indeterminate' : false)}
+            onChange={s => {
+              togglePageSelected(s.target.checked);
+              setAllInQuerySelected(false);
+            }}
+          />
+          {showEnable && (
```

Comment:
I think we need something here that will always show these buttons if everything in the query is selected since we won't know if there is something on another page that is disabled/enabled